provider-service/base/pkg/server/resource/provider.go

```diff
@@ -0,0 +1,64 @@
+package resource
+
+import (
+	"encoding/json"
+	"github.com/sky-uk/kfp-operator/apis"
+	pipelines "github.com/sky-uk/kfp-operator/apis/pipelines/v1alpha6"
+	"github.com/sky-uk/kfp-operator/argo/common"
+)
+
+type PipelineDefinition struct {
+	Name          common.NamespacedName `json:"name" yaml:"name"`
+	Version       string                `json:"version" yaml:"version"`
+	Image         string                `json:"image" yaml:"image"`
+	TfxComponents string                `json:"tfxComponents" yaml:"tfxComponents"`
+	Env           []apis.NamedValue     `json:"env" yaml:"env"`
+	BeamArgs      []apis.NamedValue     `json:"beamArgs" yaml:"beamArgs"`
+	Manifest      json.RawMessage       `json:"manifest,omitempty"`
```

Comment:
Why is this a raw message? According to the job enricher logic, we're expecting particular fields to be present in this json so it would make sense that this is reflected in the type definition.